sklearn/ensemble/iforest.py

```diff
@@ -0,0 +1,245 @@
+# Authors: Nicolas Goix <nicolas.goix@telecom-paristech.fr>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+# License: BSD 3 clause
+
+from __future__ import division
+
+import numpy as np
+from warnings import warn
+
+from scipy.sparse import issparse
+
+from ..externals.joblib import Parallel, delayed
+from ..tree import ExtraTreeRegressor
+from ..tree._tree import DTYPE
+from ..utils import check_random_state, check_array
+
+from .bagging import BaseBagging
+from .forest import _parallel_helper
+from .base import _partition_estimators
+
+__all__ = ["IsolationForest"]
+
+
+class IsolationForest(BaseBagging):  # code structure from RandomTreesEmbedding
+    """Isolation Forest Algorithm
+
+    Return the anomaly score of each sample with the IsolationForest algorithm
+
+    IsolationForest consists in 'isolate' the observations by randomly
+    selecting a feature and then randomly selecting a split value
+    between the maximum and minimum values of the selected feature.
+
+    Since recursive partitioning can be represented by a tree structure, the
+    number of splitting required to isolate a point is equivalent to the path
+    length from the root node to a terminating node.
+
+    This path length, averaged among a forest of such random trees, is a
+    measure of abnormality and our decision function.
+
+    Indeed random partitioning produces noticeable shorter paths for anomalies.
+    Hence, when a forest of random trees collectively produce shorter path
+    lengths for some particular points, then they are highly likely to be
+    anomalies.
+
+
+    Parameters
+    ----------
+    n_estimators : int, optional (default=100)
+        The number of base estimators in the ensemble.
+
+    max_samples : int or float, optional (default=256)
```

Comment:
Yes it is ! That is why we needed the _fit method: if `n_samples < 256`, `max_samples` has to be changed in the fit which is not allowed.